src/common/upload.ts

```diff
@@ -297,4 +299,11 @@ export abstract class TurboAuthenticatedBaseUploadService
       manifestResponse,
     };
   }
+
+  async getNativeAddress(): Promise<NativeAddress> {
+    return ownerToNativeAddress(
+      toB64Url(await this.signer.getPublicKey()),
+      this.token,
+    );
```

Comment:
Consideration (a bit of thinking out loud): make `getNativeAddress` a method on the signer interface vs here. Each signer would implement the logic to derive owner instead of using the utility function. TurboFactory can then expose the signer public ally so a client could call it like `turbo.signer.getNativeAddress`.

What do you think? It seems odd to leverage a utility class with a bunch of switch statements. I can see there being issue with Kyve vs. Eth, but maybe they are subclasses of an abstract Eth Signer that implement specific APIs necessary.

Reply:
[9b4d373](https://github.com/ardriveapp/turbo-sdk/pull/142/commits/9b4d3736dfa950a57486f3e52c0be81c117b8ecf)

Reply:
also moved the util: 6ee3b00

each token specific signer right now is within the arbundles layer. we could move that logic into the signer in a fork. but for now we can have our TurboDataItemSigner hold this logic.